devito/ops/operator.py

```diff
@@ -37,7 +36,10 @@ def _specialize_iet(self, iet, **kwargs):
 
         # If there is no affine trees, then there is no loop to be optimized using OPS.
         if not affine_trees:
-            return iet
+            self._should_compile_with_ops = False
```

Comment:
this flag is a hack, I'd guess? If you had a tree with some loops that can be offloaded, and some loops that cannot, what would you do with it?

Reply:
if a tree has at least one loop that can be offloaded, then this will be enough for building a list of kernels to be ops`fied. 
therefore, there will be a `self.hcode`, and thus there will be no problem calling `self._compiler.jit_compile(self._soname, str(self.ccode), str(self.hcode))`.

only if there aren't any loops at all that can be offloaded there will be no `str(self.hcode)` to serve as the 3rd param for ops` `jit_compile`

Reply:
If you support the _general_ case of an Opertor that contains **both** an affine and a number of non-affine nests, then I don't see why in practice you shouldn't be able to support the two special cases

* only affine nests are present
* only non-affine (ie non-offloadable) nests are present

Reply:
if you run a plain C code through the OPS compiler/code generator, w/o any of the OPS constructs, it should just produce/return the same input code, unmodified